test/test_sendrecv.jl

```diff
@@ -132,5 +132,7 @@ MPI.Sendrecv!(a, dest_rank, 2,
 
 @test b == [(comm_rank+1) % comm_size, (comm_rank+1) % comm_size, (comm_rank+1) % comm_size]
 
+@test MPI.Waitall!([]) == MPI.Status[]
```

Comment:
I think you might need
```suggestion
@test MPI.Waitall!(Request[]) == MPI.Status[]
```